repo/packages/C/cassandra/0/config.json

```diff
@@ -1,104 +1,135 @@
 {
   "type": "object",
   "properties": {
-    "mesos/master": {
-      "description": "The URL of the Mesos master. The format is a comma-delimited list of hosts like zk://host1:port,host2:port/mesos. If using ZooKeeper, pay particular attention to the leading zk:// and trailing /mesos! If not using ZooKeeper, standard URLs like http://localhost are also acceptable.",
-      "type": "string",
-      "default": "zk://master.mesos:2181/mesos"
-    },
-
-    "framework/failover-timeout-seconds": {
-      "description": "The failover_timeout for Mesos in seconds. If the framework instance has not re-registered with Mesos this long after a failover, Mesos will shut down all running tasks started by the framework.",
-      "type": "integer",
-      "minimum": 0,
-      "default": 604800
-    },
-    "framework/role": {
-      "description": "Mesos role for this framework.",
-      "type": "string",
-      "default": "*"
-    },
-    "framework/authentication/enabled": {
-      "description": "Whether framework authentication should be used",
-      "type": "boolean",
-      "default": false
-    },
-    "framework/authentication/principal": {
-      "description": "The Mesos principal used for authentication.",
-      "type": "string"
-    },
-    "framework/authentication/secret": {
-      "description": "The path to the Mesos secret file containing the authentication secret.",
-      "type": "string"
-    },
+    "mesos": {
+      "description": "Mesos specific configuration properties",
+      "type": "object",
+      "properties": {
+        "master": {
+          "default": "zk://master.mesos:2181/mesos",
+          "description": "The URL of the Mesos master. The format is a comma-delimited list of hosts like zk://host1:port,host2:port/mesos. If using ZooKeeper, pay particular attention to the leading zk:// and trailing /mesos! If not using ZooKeeper, standard host:port patterns, like localhost:5050 or 10.0.0.5:5050,10.0.0.6:5050, are also acceptable.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "master"
+      ]
+    },
+    "cassandra": {
+      "description": "Cassandra Framework Configuration Properties",
+      "type": "object",
+      "properties": {
+        "framework": {
+          "description": "Framework Scheduler specific Configuration Properties",
+          "type": "object",
+          "properties": {
+            "failover-timeout-seconds": {
+              "description": "The failover_timeout for Mesos in seconds. If the framework instance has not re-registered with Mesos this long after a failover, Mesos will shut down all running tasks started by the framework.",
+              "type": "integer",
+              "minimum": 0,
+              "default": 604800
+            },
+            "role": {
+              "description": "Mesos role for this framework.",
+              "type": "string",
+              "default": "*"
+            },
+            "authentication": {
```

Comment:
I'd recommend setting `additionalProperties` to false, at least for this nested object.  That will make typos fail the validation step.  Imagine you had mis-spelled `enabled`...